### ISSUE TITLE
AWS RDS IAM authentication fails

### ISSUE THREAD

**Issue by romo-dw** [aws/aws-toolkit-jetbrains#2124]
I was glad to see that IAM authentication recently got supported and GUI seems pretty straightforward so I don't know what I'm missing.

**Describe the bug**
When trying to connect through my SSM/SSH tunnel, everything's quiet for 60 seconds and then I get access denied.

However, things are working if I run `mysql`:

```
mysql --host=127.0.0.1 --port=5435 --user=$USERNAME --password=$PASSWORD --enable-cleartext-plugin
```

... where `$PASSWORD` is a generated auth token using `aws`.

**To reproduce**
1. Install `aws-toolkit`
2. Add new data source with driver `Amazon Aurora MySQL`
3. Enter host/port of tunnel
4. Set authentication to `AWS IAM`
5. Enter database username
6. Select credentials (I use SSO - if that matters)
7. Select region that database is running in
8. Enter database hostname and port
9. Hit the "Test Connection" button

**Expected behavior**
A connection is successfully made.

**Screenshots**
![data-sources-and-drivers-denied](https://user-images.githubusercontent.com/24540755/94996714-8c362200-05a6-11eb-8f8d-b3d318445ad4.png)

**Your Environment**
- OS: Pop!_OS 20.04 LTS
- JetBrains product: DataGrip
- JetBrains product version: 2020.2.3 Build #DB-202.7319.56
- AWS Toolkit version: 1.18-202
- SAM CLI version: 1.4.0
- JVM/Python version: 1.8.0_265 / 3.8.2

**Additional context**
I can list my S3 buckets in the "AWS Explorer" so my credentials are definitely working.

**Comment by hunterwerlla** [comment]
@romo-dw A few thoughts:
 1. Can you use the explorer action to create the connection instead? RDS recommends using the MariaDB driver, which we configure with the action: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Connecting.html#Aurora.Connecting.AuroraMySQL
2. Were you able to connect to the DB from Datagrip using a temporary or long lived password before?
3. In the SSH/SSL tab, is anything enabled? 
4. Can you verify <my user> is a valid IAM user in the database created using [these instructions](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html)

**Comment by romo-dw** [comment]
Hello @hunterwerlla 

1. Looks like RDS is `(empty)` in the AWS Explorer. How odd? I have a few different clusters in us-east-1.

2. I haven't actually used DataGrip for this use case before, only mysql CLI client. However, I just tried the MySQL driver in DataGrip with username and password (my generated auth token) and it's working fine. But if I switch to Aurora MySQL driver with username and password (not IAM auth!) it does not work.

3. Nothing enabled there.

4. I see my user there. `AWSAuthenticationPlugin` set for that one. Nothing in particular jumps out.

**Comment by hunterwerlla** [comment]
If RDS is empty in the AWS Explorer, are you using the right credential profile? That seems like the most likely issue.

**Comment by romo-dw** [comment]
Yes, I'm using the right credential profile. I can see all my S3 buckets that belong to that account, but I don't see any RDS instances.

**Comment by hunterwerlla** [comment]
@romo-dw Does it say "no resources" or "Failed to load resources"?

**Comment by romo-dw** [comment]
This is what I see:

![Screenshot from 2020-10-07 21-23-32](https://user-images.githubusercontent.com/24540755/95377874-6cbf3200-08e3-11eb-84bb-d1ecc96321b9.png)

I took a look at the DataGrip logs and I see nothing of particular value except maybe this warning (probably not relevant though?):

`WARN - oncurrency.BoundedTaskExecutor - Pool name must be capitalized but got: '0.Invoker.Background.Thread: software.aws.toolkits.jetbrains.core.explorer.AwsExplorerTreeStructure@7220d1ba @ ExplorerToolWindow.kt : 74'`

I don't see anything related to "no resources" or "Failed to load resources" in the GUI or in the log.

**Comment by hunterwerlla** [comment]
@romo-dw Can you check that the region is correct as well? if it says `empty` that means the list worked and returned nothing, so there are definitely no databases in that region.

**Comment by romo-dw** [comment]
I can confirm it's the right region. 100 % certain. If I change my region to `us-east-2`, the list of CloudFormation stacks and Lambdas in `us-east-1` disappears and I find my stuff from `us-east-2`. If I switch back then I see my stuff from `us-east-1` again but RDS still says `empty`.

**Comment by hunterwerlla** [comment]
@romo-dw when you run `aws rds describe-db-instances`, does the db show up? That is the exact API call we make from the explorer.

**Comment by romo-dw** [comment]
@hunterwerlla Yes, all my databases show up like expected when I'm using the `aws` cli.

**Comment by abrooksv** [comment]
Looks like the engine name for Aurora mysql is wrong. We have it as `aurora`, but its `aurora-mysql`.

This causes the explorer to return empty.

**Comment by romo-dw** [comment]
@abrooksv I'm using Aurora MySQL 5.7. That might be the issue.

**Comment by hunterwerlla** [comment]
@romo-dw We were finally able to reproduce the issue, using the `aurora mysql` driver seemed to cause the issue, using `mysql` worked, can you try that?

We will also fix Aurora MySQL 5.7+ showing up in the explorer, thanks for reporting that as well!

**Comment by romo-dw** [comment]
@hunterwerlla Great news! I actually tried the `mysql` driver in DataGrip last week (second point: https://github.com/aws/aws-toolkit-jetbrains/issues/2124#issuecomment-705050920) so I do know that it works. It was just `aurora mysql` that was not working.

**Comment by hunterwerlla** [comment]
@romo-dw Based on your feedback we fixed several bugs in the 1.20 release: Aurora 5.7 should show up now, and the `mysql aurora` driver should work with IAM auth. Please try it out again, and thanks for the bug feedback!
